architecture/adr-list/adr-002-identity-transactions.md

```diff
@@ -0,0 +1,272 @@
+      "type": "Ed25519VerificationKey2020", // external (property value)
+      "controller": "N22KY2Dyvmuu2PyyqSFKue",
+      "publicKeyMultibase": "z5TVraf9itbKXrRvt2DSS95Gw4vqU3CHAdetoufdcKazA"
+    }
+  ]
+ }
+```
+
+#### Verification Method
+
+1. **`id`** (string): A string with format `<DIDDoc-id>#<key-alias`
+2. **`controller`**: A list of base58-encoded identifier strings.
+3. **`type`** (string)
+4. **`publicKeyJwk`** (`map[string,string]`, optional): A map representing a JSON Web Key that conforms to [RFC7517](https://tools.ietf.org/html/rfc7517). See definition of `publicKeyJwk` for additional constraints.
+5. **`publicKeyMultibase`** (optional): A base58-encoded string that conforms to a [MULTIBASE](https://datatracker.ietf.org/doc/html/draft-multiformats-multibase-03) encoded public key.
+
+**Example:**
+
+```javascript
+{
+  "id": "N22KY2Dyvmuu2PyyqSFKue#key-0",
+  "type": "JsonWebKey2020",
+  "controller": "N22KY2Dyvmuu2PyyqSFKue",
+  "publicKeyJwk": {
+    "kty": "OKP", // external (property name)
+    "crv": "Ed25519", // external (property name)
+    "x": "VCpo2LMLhn6iWku8MKvSLg2ZAoC-nlOyPVQaO3FxVeQ" // external (property name)
+}
+```
+
+#### Service
+
+1. **`id`** (string): The value of the id property MUST be a URI conforming to [RFC3986](https://www.rfc-editor.org/rfc/rfc3986). A conforming producer MUST NOT produce multiple service entries with the same ID. A conforming consumer MUST produce an error if it detects multiple service entries with the same ID. It has a follow format: `<DIDDoc-id>#<service-alias>`
+2. **`type`** (string): The service type and its associated properties SHOULD be registered in the DID Specification Registries [DID-SPEC-REGISTRIES](https://www.w3.org/TR/did-spec-registries/)
+3. **`serviceEndpoint`** (strings): A string that conforms to the rules of [RFC3986](https://www.rfc-editor.org/rfc/rfc3986) for URIs, a map, or a set composed of a one or more strings that conform to the rules of [RFC3986](https://www.rfc-editor.org/rfc/rfc3986) for URIs and/or maps.
+
+**Example:**
+
+ ```javascript
+"service": [{
+  "id":"N22KY2Dyvmuu2PyyqSFKue#linked-domain",
+  "type": "LinkedDomains",
+  "serviceEndpoint": "https://bar.example.com"
+}]
+```
+
+#### Update `DID`
+
+If there is no DID transaction with the specified DID \(`dest`\), it is considered as a creation request for a new DID.
+
+If there is a DID transaction with the specified DID \(`dest`\), then this is update of existing DID. In this case, we can specify only the values we would like to override. All unspecified values remain the same. E.g., if a key rotation needs to be performed, the owner of the DID needs to send a DID transaction request with `dest`, `verkey` only. `alias` will stay the same.
+
+**Note**: Fields `dest` and `owner` should have the same value.
+
+#### State format
+
+`id -> {encode(data, creators), tx_hash, tx_timestamp }`
+
+### `SCHEMA`
+
+This transaction is used to create a Schema associated with credentials.
+
```

Comment:
A schema *may* be stored on ledger but it doesn't have to be. For instance, it could be stored as a schema.org URL. Question is...from a wider Evernym perspective, do we want to be flexible here? For instance, on ledger we could allow for this to be either on ledger or somewhere else, and expect the VDR Tools SDK to always read/write from ledger.

Reply:
My opinion, that the main feature of Cheqd it's an opportunity of storing everything for credential issuing in the cheqd ledger. And the killer feature is an opportunity to use coins, DID, credential issuing in one place.
But of course it's a question for Product Owners of Evernym and Cheqd.

Anyway, if we don't want to store a schema in the ledger in the future, it shouldn't affect this transaction.

Reply:
It might impact the format of what's stored, in case the schema is an HTTP URL, like https://schema.org/docs/full.html

So if this is implement as a DIDDoc, it's more extensible.

Reply:
did:cheqd:mainnet:identifier <- DIDDoc with JSON-LD description of the schema. Discussed with @Toktar, needs follow-up with Brent/Richard/Sergey K.